packages/forms/src/UIForm/UIForm.component.js

```diff
@@ -44,13 +43,37 @@ export default class UIForm extends React.Component {
 	 * @param value The payload new value
 	 */
 	onChange(event, { schema, value }) {
+		console.log('##########@');
 		const payload = {
 			formName: this.props.formName,
 			properties: this.props.properties,
 			schema,
 			value,
+			formData: mutateValue(this.props.properties, schema.key, value),
 		};
-		this.props.onChange(event, payload);
+		if (this.props.moz) {
+			this.props.onChange(payload);
+		} else {
+			this.props.onChange(event, payload);
+		}
+		// moving trigger in onChange implies that trigger will be triggered even if value is not valid
+		// onChange/onFinish separation was to allow widgets and fields to manage when those are called. Some widgets call onChange then onFinish on the same event.
+		// We can for example on selection widget, to trigger both, but on user typing widgets, call onFinish on blur
+		// trigger if value is correct
+		// if (schema.triggers && schema.triggers.length) {
```

Comment:
µYou can keep the comment explaining why we don't have trigger here, but remove the commented code please